hxinsight-extension/src/main/java/org/alfresco/hxi_connector/hxi_extension/service/HxInsightClient.java

```diff
@@ -168,4 +173,28 @@ public String retryQuestion(String questionId, String comments, Question questio
                 .build());
         return askQuestion(question);
     }
+
+    public BinaryResource getAvatar(String agentId)
+    {
+        try
+        {
+            HttpRequest request = HttpRequest.newBuilder()
+                    .uri(URI.create(format(config.getAvatarUrl(), agentId)))
+                    .headers(authService.getAuthHeaders())
+                    .GET()
+                    .build();
+
+            HttpResponse<InputStream> httpResponse = client.send(request, BodyHandlers.ofInputStream());
+
+            ensureCorrectHttpStatusReturned(SC_OK, httpResponse);
+            log.atDebug().log("Agent with id {} received avatar successfully", agentId);
+
+            File tempImageFile = TempFileProvider.createTempFile(httpResponse.body(), format("avatar-%s", agentId), "png");
```

Comment:
What happens here if we request multiple times the avatar for the same agentId? Does the `TempFileProvider` implementation include randomness? Will we have a conflict on trying to store the same file twice? Does the implementation override the previous file or leave the previous one, if no randomness is there?

Reply:
I changed the file prefix to be unique.